bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -970,4 +970,162 @@ mod tests {
         init::init();
         assert!(init::fips_mode().unwrap().is_enabled());
     }
+
+    /// Test that an application context can be set and retrieved.
+    #[test]
```

Comment:
Can you put most of these tests in connection.rs? They don't need any of the testing framework functionality, bc they don't do handshakes or IO (except test_app_context_callback).